pkg/controllers/management/feature/feature.go

```diff
@@ -0,0 +1,56 @@
+package feature
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/rancher/rancher/pkg/clustermanager"
+	"github.com/rancher/rancher/pkg/features"
+	v3 "github.com/rancher/types/apis/management.cattle.io/v3"
+	"github.com/rancher/types/config"
+	"github.com/sirupsen/logrus"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+type featuresController struct {
+	features v3.FeatureInterface
+}
+
+func RegisterEarly(ctx context.Context, management *config.ManagementContext, clusterManager *clustermanager.Manager) {
+	f := &featuresController{
+		management.Management.Features(""),
+	}
+
+	f.SetProvider(management)
+
+	management.Management.Features("").AddHandler(ctx, "feature-settings", f.sync)
+}
+
+func (f *featuresController) sync(key string, obj *v3.Feature) (runtime.Object, error) {
+	if !features.FeatureExists(key) {
+		return nil, nil
+	}
+
+	enabled := obj.Default
+	if obj.Value != nil {
+		enabled = *obj.Value
+	}
+
+	if err := features.SetGate(key, enabled); err != nil {
+		return nil, fmt.Errorf("could not set feature setting %s: %v", key, err)
+	}
+
+	return nil, nil
+}
+
+func (f *featuresController) SetProvider(management *config.ManagementContext) {
```

Comment:
Get features off management, and doesn't need to live off the featuresController struct. And doesn't need to be exported.